### DISCUSSION TITLE
How to group with different scale class?

### DISCUSSION THREAD

**Discussion by mqc007** [neperfepx/neper#472] [Q&A] [answered]
Hi, 

I am using the neper to generate two-scales tess. and I am wondering how to visualize the model with diferent classes. like the first class with different colors and the second class with the same color. 

here is the code that  i used:

`neper -T -n 144::10 -domain "cube(10,10,10)" -reg 1 -morpho gg -format tess,3dec -o 0510`

I am not sure the "class" is right, but I assuming the first I need to do is to group second class blocks, so could you please show me how to group second class blocks? 

another question is when import the tess into 3dec, some blocks are too small to generate mesh, is there any suggestion to avoid this case? Like if i can set the min size of the block when i generating the tess? Or is that case because of the growgrain model? Or is that because i divided too much? 

look forward to hearing from you.

many thanks
regards

**Comment by rquey** [comment, marked as answer]
> Hi,
> 
> I am using the neper to generate two-scales tess. and I am wondering how to visualize the model with diferent classes. like the first class with different colors and the second class with the same color.
> 
> here is the code that i used:
> 
> `neper -T -n 144::10 -domain "cube(10,10,10)" -reg 1 -morpho gg -format tess,3dec -o 0510`
> 
> I am not sure the "class" is right, but I assuming the first I need to do is to group second class blocks, so could you please show me how to group second class blocks?

Use `-datacellcol "scaleid(1)"`:
```
neper --rcfile none -V 0510.tess -print img1
neper --rcfile none -V 0510.tess -datacellcol "scaleid(1)" -print img2
```
![image](https://user-images.githubusercontent.com/2659376/167851595-06d5e00a-fd57-433d-a34d-ab3e0d32af30.png)
![image](https://user-images.githubusercontent.com/2659376/167851615-3c75348d-b023-42c6-9e88-cc4f0964088d.png)

> another question is when import the tess into 3dec, some blocks are too small to generate mesh, is there any suggestion to avoid this case? Like if i can set the min size of the block when i generating the tess? Or is that case because of the growgrain model? Or is that because i divided too much?

This is probably a 3dec issue, not a Neper issue.

**Reply by mqc007** [reply]
Thank you so much!
